primitives/core/src/network.rs

```
@@ -0,0 +1,103 @@
+// Copyright 2020-2023 Litentry Technologies GmbH.
+// This file is part of Litentry.
+//
+// Litentry is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Litentry is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Litentry.  If not, see <https://www.gnu.org/licenses/>.
+
+use codec::{Decode, Encode, MaxEncodedLen};
+use scale_info::TypeInfo;
+use sp_runtime::{traits::ConstU32, BoundedVec};
+use sp_std::{hash::Hash, vec::Vec};
+use strum::IntoEnumIterator;
+use strum_macros::EnumIter;
+
+pub const MAX_WEB3NETWORK_LEN: u32 = 128;
+pub type BoundedWeb3Network = BoundedVec<Web3Network, ConstU32<MAX_WEB3NETWORK_LEN>>;
+
+/// supported web3 networks
+/// use a flattened style to avoid overly nested structure like:
+/// {
+///   Substrate(SubstrateNetwork),
+///   Evm(EvmNetwork),
+/// }
+/// TODO: theoretically this should the the union of the supported networks of all data providers
+#[derive(
+	Encode,
+	Decode,
+	Copy,
+	Clone,
+	Debug,
+	PartialEq,
+	Eq,
+	PartialOrd,
+	Ord,
+	Hash,
+	TypeInfo,
+	MaxEncodedLen,
+	EnumIter,
+)]
+pub enum Web3Network {
+	// substrate
+	Polkadot,
+	Kusama,
+	Litentry,
+	Litmus,
+	LitentryRococo,
+	Khala,
+	SubstrateTestnet, // when launched it with standalone (integritee-)node
+
+	// evm
+	Ethereum,
+	Polygon,
+	BSC,
+}
+
+impl Web3Network {
+	pub fn is_substrate(&self) -> bool {
+		matches!(
```

Have you considered using the normal match as it's exhaustive and requires you handling a new variant. Adding following test could also prevent from such mistakes:

```
	#[test]
	fn is_evm_network() {
		Web3Network::iter().for_each(|network| {

			assert_eq!(network.is_evm(), match network {
				Web3Network::Polkadot => false,
				Web3Network::Kusama => false,
				Web3Network::Litentry => false,
				Web3Network::Litmus => false,
				Web3Network::LitentryRococo => false,
				Web3Network::Khala => false,
				Web3Network::SubstrateTestnet => false,
				Web3Network::Ethereum => true,
				Web3Network::Polygon => true,
				Web3Network::BSC => true
			})
		})
	}
```

---

That's a good point - I'll add test cases to prevent errors due to being silently ignored